src/nbl/video/CVulkanLogicalDevice.h

```diff
@@ -794,8 +973,49 @@ class CVulkanLogicalDevice final : public ILogicalDevice
 
     core::smart_refctd_ptr<IGPUFramebuffer> createGPUFramebuffer_impl(IGPUFramebuffer::SCreationParams&& params) override
     {
-        // Todo(achal): Hoist creation out of constructor
-        return nullptr; // return core::make_smart_refctd_ptr<CVulkanFramebuffer>(this, std::move(params));
+        // This flag isn't supported until Vulkan 1.2
+        // assert(!(m_params.flags & ECF_IMAGELESS_BIT));
+
+        constexpr uint32_t MemSize = 1u << 12;
+        constexpr uint32_t MaxAttachments = MemSize / sizeof(VkImageView);
+
+        VkImageView vk_attachments[MaxAttachments];
+        uint32_t attachmentCount = 0u;
+        for (uint32_t i = 0u; i < params.attachmentCount; ++i)
+        {
+            if (params.attachments[i]->getAPIType() == EAT_VULKAN)
```

Comment:
you need to hardcore assert this, or bail out (retur nullptr) on the first one that doesnt fullfill this

otherwise the indices to the attachments in the subpass dependencies, etc. will all be messed up